hack/git/hooks/pre-commit

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+set -e
+LC_ALL=C
+
+git_username_lower="$(git config github.user | tr '[:upper:]' '[:lower:]')"
+if [[ -z "${git_username_lower}" ]]
+then
+    echo "Please set github.user (git config github.user) locally or globally before issuing commits to this repo."
+    exit 1
+fi
+
+# e.g. "USERNAME/ARO-1234", "USERNAME/hotfix-v20240321.00", or "USERNAME/gh-issue-123"
+valid_branch_regex="^${git_username_lower}\/(ARO-[0-9]{4}[a-z0-9._-]*|hotfix-[a-z0-9._-]+|gh-issue-[0-9]+[a-z0-9._-]*)$"
```

Comment:
Minor objection to the `$` at the end.  Could we allow for an optional 3rd free-form segment for cards/issues requiring multiple PRs.  Like:
```
USERNAME "/" TICKET ["/" .*]
```

I'm just imagining a followup PR for something that was missed or deferred in the initial review.  e.g. `mbarnes/ARO-1234/bugfix`

Reply:
is `ARO-1234-bugfix` not acceptable in this case? I can add `/` to the regex if that is considered a blocker.

Reply:
I think, for our goals with naming consistency here, we should only really need to enforce the prefix of the name, if all the required information is present at the start of the name for easy identification, it shouldn't matter what other suffixes we might need to include on the branch, be it identifications for split units of work, easy clarification to identify branches without memorizing Jira IDs, etc.

Reply:
My bad, I didn't read the regex close enough.  A 2nd slash would make the branch name easier to parse if there's ever a need to do that.  But not important enough to block on.